File: tasks/2023/final/snacks.py
#!/usr/bin/env python3
# @check-accepted: *

n, x = map(int, input().split())
l = list(map(int, input().strip().split()))

l.sort()
t = 0
i = 0
while True:
    if i + 1 < n and l[i] + l[i + 1] <= x:
        i += 2
    elif i < n and l[i] <= x:
        i += 1
    else:
        break
    t += 1

print(t)

File: tasks/2023/final/snacks.cpp
// @check-accepted: *

#include <algorithm>
#include <iostream>
#include <vector>

int main() {
    int n, x;
    std::cin >> n >> x;

    std::vector<int> l(n);
    for (int i = 0; i < n; i++) {
        std::cin >> l[i];
    }
    sort(l.begin(), l.end());

    int t = 0;
    for (int i = 0;; ++t) {
        if (i + 1 < n && l[i] + l[i + 1] <= x) {
            i += 2;
        } else if (i < n && l[i] <= x) {
            i += 1;
        } else {
            break;
        }
    }

    std::cout << t << '\n';
}
